client/src/href-builder.js

```diff
@@ -34,8 +34,8 @@ module.exports = function(config){
           registryUrl = clientRenderingEndpoint,
           registrySegment = registryUrl.slice(-1) === '/' ? registryUrl : (registryUrl + '/'),
           qs = !!component.parameters ? ('/?' + querystring.stringify(component.parameters)) : '';
-
-      return url.resolve(registrySegment, component.name) + versionSegment + qs;
+      
+      return url.resolve(registrySegment, component.name || '') + versionSegment + qs;
```

Comment:
Can we actually validate component.name is not empty here and in case return a validation error? No point of making a req if component.name is missing.